db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/task/DailyTest.java

```diff
@@ -5,47 +5,138 @@
-		return ExecutionComplete.success(null, timeDone, timeDone);
-	}
-}
\ No newline at end of file
+    private static final ZoneId ZONE = ZoneId.systemDefault();
+    private static final ZoneId LONDON = ZoneId.of("Europe/London");
+    private static final ZoneId UTC = ZoneId.of("UTC");
+    private static final ZoneId ROME = ZoneId.of("Europe/Rome");
+
+    private static final LocalTime HOUR_0 = LocalTime.of(0, 0);
+    private static final LocalTime HOUR_1 = LocalTime.of(1, 0);
+    private static final LocalTime HOUR_2 = LocalTime.of(2, 0);
+    private static final LocalTime HOUR_3 = LocalTime.of(3, 0);
+    private static final LocalTime HOUR_6 = LocalTime.of(6, 0);
+    private static final LocalTime HOUR_7 = LocalTime.of(7, 0);
+    private static final LocalTime HOUR_8 = LocalTime.of(8, 0);
+    private static final LocalTime HOUR_12 = LocalTime.of(12, 0);
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void should_not_allow_empty_times() {
+        expectedException.expect(IllegalArgumentException.class);
+        new Daily();
+    }
+
+    @Test
+    public void should_not_allow_null_zone_id() {
+        expectedException.expect(NullPointerException.class);
+        new Daily(null, new LocalTime[]{LocalTime.MIDNIGHT});
+    }
+
+    @Test
+    public void should_generate_next_date_correctly() {
+        LocalDate currentDate = Instant.now().atZone(ZONE).toLocalDate();
+        LocalDate nextDay = currentDate.plusDays(1);
+
+        assertThat(new Daily(HOUR_8).getNextExecutionTime(complete(instant(currentDate, HOUR_0))), is(instant(currentDate, HOUR_8)));
+        assertThat(new Daily(HOUR_8).getNextExecutionTime(complete(instant(currentDate, HOUR_8))), is(instant(nextDay, HOUR_8)));
+
+        assertThat(new Daily(HOUR_8, HOUR_12).getNextExecutionTime(complete(instant(currentDate, HOUR_0))), is(instant(currentDate, HOUR_8)));
+        assertThat(new Daily(HOUR_8, HOUR_12).getNextExecutionTime(complete(instant(currentDate, HOUR_8))), is(instant(currentDate, HOUR_12)));
+        assertThat(new Daily(HOUR_8, HOUR_12).getNextExecutionTime(complete(instant(currentDate, HOUR_12))), is(instant(nextDay, HOUR_8)));
+        //order should be irrelevant
+        assertThat(new Daily(HOUR_12, HOUR_8).getNextExecutionTime(complete(instant(currentDate, HOUR_0))), is(instant(currentDate, HOUR_8)));
+    }
+
+    @Test
+    public void should_generate_next_date_correctly_during_standard_time() {
+        // Europe/Rome is UTC+1 during standard time
+        LocalDate currentDate = LocalDate.of(2020, 12, 12);
+        LocalDate nextDay = currentDate.plusDays(1);
+
+        assertThat(new Daily(ROME, HOUR_8).getNextExecutionTime(complete(instant(currentDate, HOUR_0, UTC))), is(instant(currentDate, HOUR_7, UTC)));
+        assertThat(new Daily(ROME, HOUR_8).getNextExecutionTime(complete(instant(currentDate, HOUR_7, UTC))), is(instant(nextDay, HOUR_7, UTC)));
+    }
+
```

Comment:
Good job adding all the tests for the edge-cases 👍

Reply:
Thank you!